python_modules/libraries/dagster-dg/dagster_dg/context.py

```diff
@@ -417,10 +417,14 @@ def external_components_command(self, command: list[str], log: bool = True) -> s
                 return result.stdout.decode("utf-8")
 
     def ensure_uv_lock(self, path: Optional[Path] = None) -> None:
+        path = path or self.root_path
+        if not (path / "uv.lock").exists():
+            self.ensure_uv_sync(path)
+
+    def ensure_uv_sync(self, path: Optional[Path] = None) -> None:
         path = path or self.root_path
         with pushd(path):
```

Comment:
we could add a cli log line here to indicate that we are syncing the project if its not clear from what uv already outputs?